client/packages/common/src/intl/utils/DateUtils.ts

```diff
@@ -60,6 +60,18 @@ const formatIfValid = (
   }
 ): string => (isValid(date) ? format(date, dateFormat, options) : '');
 
+// Adds the current time to a date object (that presumably has 00:00 as its time
+// component) -- does not mutate input Date object)
+const addCurrentTime = (date: Date | null): Date | null => {
+  if (date === null) return date;
+  const d = new Date();
+  const e = new Date(d);
+  const msSinceMidnight = d.getTime() - e.setHours(0, 0, 0, 0);
+  const newDate = new Date(date);
+  newDate.setTime(e.getTime() + msSinceMidnight);
```

Comment:
But `e` has the timestamp of  `now -  msSinceMidnight` and not of the input `date` and you then set newDate to `now - msSinceMidnight + msSinceMid = now`, or do I miss something? 

think you have to use `newDate.setHours(0, 0, 0, 0)`

Reply:
Ah yes, that's right, got mixed up with which was which. Fixed now: 7c46dfb59